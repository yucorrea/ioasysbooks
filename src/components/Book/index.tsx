import React, { memo } from 'react';
import styled from 'styled-components/native';
import { TouchableOpacity, TouchableOpacityProps } from 'react-native';
import Animated, { LightSpeedInLeft } from 'react-native-reanimated';
import { Book as BookData } from '@routes/AppStack';


const AnimatedStyledContainer = Animated.createAnimatedComponent(TouchableOpacity);
interface Props extends TouchableOpacityProps {
  data: BookData;
  onPress: () => void
}

const Book : React.FC<Props> = ({ data, onPress, ...rest })  => {

  return (
    <StyledContainer
      entering={LightSpeedInLeft}
      onPress={onPress}
      {...rest}>
      <StyledImage source={{ uri: data.imageUrl }} />
      <StyledWrapper>
        <StyledTitle>{data.title}</StyledTitle>
        <StyledSubTitle>{data.authors[0]}</StyledSubTitle>

        <StyledFooter>
          <StyledDetail>{data.pageCount} pÃ¡ginas</StyledDetail>
          <StyledDetail>Editora {data.publisher}</StyledDetail>
          <StyledDetail>Publicado em {data.published}</StyledDetail>
        </StyledFooter>
      </StyledWrapper>
    </StyledContainer>
  );
}

const StyledContainer = styled(AnimatedStyledContainer)`
  flex-direction: row;
  width: 100%;
  height: 160px;
  margin: 16px 0px;
  padding: 19px 16px;
  border-radius: 4px;
  background: ${({ theme }) => theme.colors.shape};
`;

const StyledImage = styled.Image`
  width: 81px;
  height: 112px;
  margin-right: 16px;
`;

const StyledWrapper = styled.View``;

const StyledTitle = styled.Text`
  color: ${({ theme }) => theme.colors.title};
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.medium};
`;

const StyledSubTitle = styled.Text`
  color: ${({ theme }) => theme.colors.primary_light};
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

const StyledFooter = styled.View`
  margin-top: 25px;
`;

const StyledDetail = styled.Text`
  color: ${({ theme }) => theme.colors.text};
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

export default memo(Book, (prevState, nextState) => {
  return Object.is(prevState.data, nextState.data);
}) ;
