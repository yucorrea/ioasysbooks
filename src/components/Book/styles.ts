import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  width: 100%;
  height: 160px;
  background: ${({ theme }) => theme.colors.background};

  flex-direction: row;
  padding: 19px 16px;
  margin: 16px 0px;

  border-radius: 4px;
`;

export const Image = styled.Image`
  width: 81px;
  height: 112px;
  margin-right: 16px;
`;

export const Wrapper = styled.View``;

export const Title = styled.Text`
  color: ${({ theme }) => theme.colors.title};
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.medium};
`;

export const SubTitle = styled.Text`
  color: ${({ theme }) => theme.colors.primary_light};
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

export const Footer = styled.View`
  margin-top: 25px;
`;

export const Detail = styled.Text`
  color: ${({ theme }) => theme.colors.text};
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;
