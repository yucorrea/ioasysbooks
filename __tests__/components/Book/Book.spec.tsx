import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import Book from '../../../src/components/Book';
import {
  matchSnapshotWithTheme,
  renderWithTheme,
} from '../../../src/helpers/functions/testing';

describe('Buttom Component', () => {

  const data = {
    id: 'book-id',
    imageUrl: 'image.png',
    title: 'Book Title',
    pageCount: 1,
    publisher: '',
    published: 0,
    isbn10: '',
    isbn13: '',
    category: '',
    authors: [],
    language: '',
    description: 'An book description',
  }

  it('matches snapshots', () => {
    const mockFunction = jest.fn();
    const snap = matchSnapshotWithTheme(<Book data={data} onPress={mockFunction} />);
    expect(snap).toMatchSnapshot();
  });

  it('render correctly', () => {
    const mockFunction = jest.fn();
    renderWithTheme(<Book testID="button-id" data={data} onPress={mockFunction}  />);
  });

  it('callback is called when book is pressed', () => {
    const mockFunction = jest.fn();
    const { getByTestId } = renderWithTheme(
      <Book testID="button-id" data={data} onPress={mockFunction}  />,
    );
    const book = getByTestId('button-id');
    fireEvent.press(book);
    expect(mockFunction).toHaveBeenCalled();
  });

});
