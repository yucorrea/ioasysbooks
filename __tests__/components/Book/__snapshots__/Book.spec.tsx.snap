// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttom Component matches snapshots 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  forwardedRef={[Function]}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "flexDirection": "row",
      "height": 160,
      "marginBottom": 16,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 16,
      "opacity": 1,
      "paddingBottom": 19,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 19,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "image.png",
      }
    }
    style={
      Array [
        Object {
          "height": 112,
          "marginRight": 16,
          "width": 81,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "Heebo-Medium",
            "fontSize": 14,
          },
        ]
      }
    >
      Book Title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#AB2680",
            "fontFamily": "Heebo-Regular",
            "fontSize": 12,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "marginTop": 25,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Heebo-Regular",
              "fontSize": 12,
            },
          ]
        }
      >
        1
         p√°ginas
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Heebo-Regular",
              "fontSize": 12,
            },
          ]
        }
      >
        Editora 
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Heebo-Regular",
              "fontSize": 12,
            },
          ]
        }
      >
        Publicado em 
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`Buttom Component matches snapshots 2`] = `undefined`;
